pkg/master/master_test.go

```diff
@@ -552,27 +552,77 @@ type FooList struct {
 }
 
 func initThirdParty(t *testing.T, version, name string) (*Master, *etcdtesting.EtcdTestServer, *httptest.Server, *assert.Assertions) {
+	return initThirdPartyMultiple(t, []string{version}, []string{name})
+}
+
+func initThirdPartyMultiple(t *testing.T, versions, names []string) (*Master, *etcdtesting.EtcdTestServer, *httptest.Server, *assert.Assertions) {
 	master, etcdserver, _, assert := newMaster(t)
-	api := &extensions.ThirdPartyResource{
-		ObjectMeta: api.ObjectMeta{
-			Name: name,
-		},
-		Versions: []extensions.APIVersion{
-			{
-				Name: version,
-			},
-		},
-	}
 	_, master.ServiceClusterIPRange, _ = net.ParseCIDR("10.0.0.0/24")
 
-	if !assert.NoError(master.InstallThirdPartyResource(api)) {
-		t.FailNow()
+	for ix := range names {
+		api := &extensions.ThirdPartyResource{
+			ObjectMeta: api.ObjectMeta{
+				Name: names[ix],
+			},
+			Versions: []extensions.APIVersion{
+				{
+					Name: versions[ix],
+				},
+			},
+		}
+		err := master.InstallThirdPartyResource(api)
```

Comment:
This unit test doesn't equal to real cases.
See [here](https://github.com/kubernetes/kubernetes/blob/ef0c9f0c5b8efbba948a0be2c98d9d2e32e0b68c/pkg/master/thirdparty_controller.go#L69-L75):
```Go
	hasResource, err := t.master.HasThirdPartyResource(rsrc)
	if !hasResource {
		return t.master.InstallThirdPartyResource(rsrc)
	}
```
The bug was caused in `HasThirdPartyResource` returning true because the API Group has been registered before despite of different resource kind.

I tried to fix it by comparing both API group and resource kind. But I got error:
```
[restful] WebService with duplicate root path detected:
```
Seems like rest path duplicate reg. Any suggestion on how to handle the install rest thing? @brendandburns 

@brendandburns 

/cc @polvi @xiang90